### PR TITLE
Prepare for dart 3.0 alpha changes: Handle SDK layout update

### PR DESCRIPTION
**Pull request by annagrin**
Handle SDK layout update:

- change min SDK constraint to `3.0.0-134.0.dev`
- update build_web_compilers to `^4.0.0`
- update build_runner to `^2.4.0`
- update SDK paths in code
- remove CI testing on stable and add testing on main

Towards: https://github.com/dart-lang/webdev/issues/1878

### CODE REVIEW

File: dwds/mono_pkg.yaml [hunk 1]
```diff
@@ -19,23 +15,23 @@ stages:
       - test: --tags=extension
       sdk:
         - dev
-        - stable
```
Comment (elliette) [inline dwds/mono_pkg.yaml:22]: What is the difference between stable and main?

File: frontend_server_client/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 3.3.0-dev
+
+- Require min dart SDK version of `3.0.0-134.0.dev`.
+- Support changes in the SDK layout for dart 3.0.
```
Comment (elliette) [inline frontend_server_client/CHANGELOG.md:4]: Nit: Dart (capitalized)

File: tool/pub_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
```
Comment (elliette) [inline tool/pub_upgrade.sh:1]: Very useful! Would it be possible to add it in separate PR though?

**Review by elliette — APPROVED** [review]
LGTM with a couple requests

File: dwds/debug_extension/pubspec.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: >-
   A chrome extension for Dart debugging.
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=3.0.0-134.0.dev <3.0.0'
```
Comment (nshahan) [inline dwds/debug_extension/pubspec.yaml:9]: Should we be raising the upper constraint to <4.0.0 or are we planning on another release before 3.0.0 stable ships?

File: dwds/debug_extension_mv3/pubspec.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ description: >-
   A Chrome extension for Dart debugging.
 
 environment:
-  sdk: '>=2.18.0 <3.0.0'
+  sdk: '>=3.0.0-134.0.dev <3.0.0'
```
Comment (nshahan) [inline dwds/debug_extension_mv3/pubspec.yaml:9]: ditto: 4.0.0?

File: dwds/mono_pkg.yaml (hunk 1 above)
Comment (nshahan) [reply dwds/mono_pkg.yaml:22]: Main is the latest bleeding edge tip of tree in the Dart repo. Stable is the latest released stable version.

File: dwds/debug_extension/pubspec.yaml [hunk 6]
```diff
@@ -18,8 +18,8 @@ dependencies:
 
 dev_dependencies:
   build: ^2.0.0
-  build_web_compilers: ^3.0.0
-  build_runner: ^2.0.6
+  build_web_compilers: ^4.0.0
+  build_runner: ^2.4.0
   built_collection: ^5.0.0
   dwds: ^11.0.0
   webdev: ^2.0.0
```
Comment (nshahan) [inline dwds/debug_extension/pubspec.yaml:25]: Do the dwds and webdev versions need a bump here?

File: dwds/debug_extension_mv3/pubspec.yaml [hunk 7]
```diff
@@ -15,10 +15,10 @@ dependencies:
 
 dev_dependencies:
   build: ^2.0.0
-  build_runner: ^2.0.6
+  build_runner: ^2.4.0
   built_collection: ^5.0.0
   built_value_generator: ^8.3.0
-  build_web_compilers: ^3.0.0
+  build_web_compilers: ^4.0.0
   dwds: ^16.0.0
```
Comment (nshahan) [inline dwds/debug_extension_mv3/pubspec.yaml:22]: bump dwds?

File: example/pubspec.yaml [hunk 8]
```diff
@@ -4,9 +4,10 @@ description: A web app example for webdev CLI.
 publish_to: none
 
 environment:
-  sdk: ">=2.18.0 <3.0.0"
+  sdk: ">=3.0.0-134.0.dev <3.0.0"
```
Comment (nshahan) [inline example/pubspec.yaml:7]: ditto: 4.0.0? I wont copy the comment to all the examples/fixtures but my question applies there as well. 😃

File: fixtures/_test/pubspec.yaml [hunk 9]
```diff
@@ -9,7 +9,7 @@ description: >-
 publish_to: none
 
 environment:
-  sdk: ">=2.18.0 <3.0.0"
+  sdk: ">=3.0.0-134.0.dev <3.0.0"
 
 dependencies:
   intl: ^0.17.0
```
Comment (nshahan) [inline fixtures/_test/pubspec.yaml:15]: Can't comment on it because it is not part of the diff but below we depend on the branch of build web compilers that runs in weak null safety mode. Does it read the weak sdk .dill files?

File: frontend_server_common/lib/src/resident_runner.dart [hunk 10]
```diff
@@ -17,10 +17,10 @@ import 'devfs.dart';
 import 'frontend_server_client.dart';
 import 'utilities.dart';
 
-final Uri platformDillUnsound =
-    Uri.file(p.join(dartSdkPath, 'lib', '_internal', 'ddc_sdk.dill'));
+final Uri platformDillUnsound = Uri.file(
+    p.join(dartSdkPath, 'lib', '_internal', 'ddc_outline_unsound.dill'));
 final Uri platformDillSound =
-    Uri.file(p.join(dartSdkPath, 'lib', '_internal', 'ddc_outline_sound.dill'));
+    Uri.file(p.join(dartSdkPath, 'lib', '_internal', 'ddc_outline.dill'));
```
Comment (nshahan) [inline frontend_server_common/lib/src/resident_runner.dart:23]: nit: these are named "platform" but they are using the "outline" .dills.

File: webdev/pubspec.yaml [hunk 11]
```diff
@@ -9,7 +9,7 @@ description: >-
 repository: https://github.com/dart-lang/webdev/tree/master/webdev
 
 environment:
-  sdk: ">=2.18.0 <3.0.0"
+  sdk: ">=3.0.0-134.0.dev <3.0.0"
```
Comment (nshahan) [inline webdev/pubspec.yaml:12]: I will point it out again here, 4.0.0?

**Review by nshahan — APPROVED** [review]
LGTM, with a few comments. Mostly questions about versions of deps and the SDK.

File: dwds/debug_extension/pubspec.yaml (hunk 4 above)
Comment (annagrin) [reply dwds/debug_extension/pubspec.yaml:9]: I am minimizing changes for this PR as it is already large:) I will update the constraint later, there is no actual need for it right now.

File: dwds/debug_extension/pubspec.yaml (hunk 6 above)
Comment (annagrin) [reply dwds/debug_extension/pubspec.yaml:25]: hmm I am not sure why the version is so low here. Maybe it is not used? @elliette do you know by any chance?

File: dwds/debug_extension_mv3/pubspec.yaml (hunk 5 above)
Comment (annagrin) [reply dwds/debug_extension_mv3/pubspec.yaml:9]: will do in a separate PR

File: dwds/debug_extension_mv3/pubspec.yaml (hunk 7 above)
Comment (annagrin) [reply dwds/debug_extension_mv3/pubspec.yaml:22]: will do, thanks!

Comment (annagrin) [reply dwds/debug_extension_mv3/pubspec.yaml:22]: ah this one has a local path override just below. The dwds version here would need to change after we publish dwds 17.0.0

File: example/pubspec.yaml (hunk 8 above)
Comment (annagrin) [reply example/pubspec.yaml:7]: I will do this in a separate pr, as I am trying to minimize changes in this one.

File: fixtures/_test/pubspec.yaml (hunk 9 above)
Comment (annagrin) [reply fixtures/_test/pubspec.yaml:15]: yes, we generate them for tests already

File: frontend_server_client/CHANGELOG.md (hunk 2 above)
Comment (annagrin) [reply frontend_server_client/CHANGELOG.md:4]: ack

File: frontend_server_common/lib/src/resident_runner.dart (hunk 10 above)
Comment (annagrin) [reply frontend_server_common/lib/src/resident_runner.dart:23]: renaming...

File: dwds/debug_extension/pubspec.yaml (hunk 6 above)
Comment (annagrin) [reply dwds/debug_extension/pubspec.yaml:25]: Anyway, it does not seem to break anything, so I will leave it as a separate issue.

File: webdev/pubspec.yaml (hunk 11 above)
Comment (annagrin) [reply webdev/pubspec.yaml:12]: not yet:) answered above.

File: frontend_server_client/CHANGELOG.md (hunk 2 above)
Comment (annagrin) [reply frontend_server_client/CHANGELOG.md]: done!

File: tool/pub_upgrade.sh [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+# Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+# for details. All rights reserved. Use of this source code is governed by a
```
[the remark this answers is not in the document]
Comment (annagrin) [reply tool/pub_upgrade.sh]: done:)

File: frontend_server_common/lib/src/resident_runner.dart (hunk 10 above)
Comment (annagrin) [reply frontend_server_common/lib/src/resident_runner.dart]: done!

**Comment by annagrin** [comment]
I addressed some comments, will addressed the rest (updating SDK constraints) in a separate PR. Thanks @nshahan and @elliette for the fast review!
